### PR TITLE
Upgrade apiserver-network-proxy to v0.0.9

### PR DESCRIPTION
**Pull request by Jefftree**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Upgrade network proxy to v0.0.9. Includes fixes for UDS socket leak and adds a health port.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by Jefftree** [comment]
Waiting for https://github.com/kubernetes/k8s.io/pull/945 to pass CI.

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -1731,7 +1731,9 @@ function prepare-konnectivity-server-manifest {
 
   params+=("--server-port=0")
   params+=("--agent-port=$1")
-  params+=("--admin-port=$2")
+
+  params+=("--health-port=$2")
```
Comment (cheftako) [inline cluster/gce/gci/configure-helper.sh:1735]: The documentation for prepare-konnectivity-server-manifest is now wrong. Can we please fix it?
# $2: value for variable "health_port"
# $3: value for variable "admin_port"

Comment (Jefftree) [reply cluster/gce/gci/configure-helper.sh:1735]: Nice catch. Fixed.

**Pull request merged**